shared/protocol/src/test/java/io/pravega/shared/protocol/netty/WireCommandsTest.java

```diff
@@ -894,14 +896,63 @@ public void testConditionalBlockEnd() throws IOException {
                 ce -> ce.getData().refCnt());
     }
 
+    @Test
+    public void testMergeSegmentsWithAttributes() throws IOException {
+        List<WireCommands.ConditionalAttributeUpdate> attributeUpdates = Arrays.asList(
+                new WireCommands.ConditionalAttributeUpdate(UUID.randomUUID(), WireCommands.ConditionalAttributeUpdate.REPLACE, 0, Long.MIN_VALUE),
+                new WireCommands.ConditionalAttributeUpdate(UUID.randomUUID(), WireCommands.ConditionalAttributeUpdate.REPLACE_IF_EQUALS, 0, Long.MIN_VALUE));
+        WireCommands.MergeSegments conditionalMergeSegments = new WireCommands.MergeSegments(l, testString1, testString2,
+                "", attributeUpdates);
+        testCommand(conditionalMergeSegments);
+        // Check the size of the ConditionalAttributeUpdate.
+        assertEquals(attributeUpdates.get(0).size(), 4 * Long.BYTES + 1);
+    }
+
+    @Data
+    public static final class MergeSegmentsV5 implements Request, WireCommand {
+        final WireCommandType type = WireCommandType.MERGE_SEGMENTS;
+        final long requestId;
+        final String target;
+        final String source;
+        @ToString.Exclude
+        final String delegationToken;
+
+        public MergeSegmentsV5(long requestId, String target, String source, String delegationToken) {
+            this.requestId = requestId;
+            this.target = target;
+            this.source = source;
+            this.delegationToken = delegationToken;
+        }
+
+        @Override
+        public void process(RequestProcessor cp) {}
+
+        @Override
+        public void writeFields(DataOutput out) throws IOException {
+            out.writeLong(requestId);
+            out.writeUTF(target);
+            out.writeUTF(source);
+            out.writeUTF(delegationToken == null ? "" : delegationToken);
+        }
+    }
+
+    @Test
+    public void testCompatibilityMergeSegmentsV5() throws IOException {
```

Comment:
This is good. FYI you could have achieved the same thing by taking the serialization of a "current" version and then trimming out the trailing bytes (i.e., trim 4 bytes from the end if you serialize 0 attribute updates). That's how we verify some of the other wire commands.

Reply:
Ok, good to know.